pdfminer/utils.py

```diff
@@ -862,3 +862,24 @@ def format_int_alpha(value: int) -> str:
 
     result.reverse()
     return "".join(result)
+
+
+def unpad_aes(padded: bytes) -> bytes:
+    """Remove block padding as described in PDF 1.7 section 7.6.2:
+
+    > For an original message length of M, the pad shall consist of 16 -
+    (M mod 16) bytes whose value shall also be 16 - (M mod 16).
+    > Note that the pad is present when M is evenly divisible by 16;
+    it contains 16 bytes of 0x10.
+    """
+    end = len(padded)
+    if end == 0:
+        return padded
+    if padded[end - 1] <= 16:
+        padding = padded[end - 1]
+        if padding <= end:
+            end -= padding
+            # Validate that it is really padding
+            if all(x == padding for x in padded[end:]):
+                return padded[:end]
+    return padded
```

Comment:
I found this difficult to understand. While understanding it, I wrote this. Do you think that is equivalent? 

```suggestion
    if len(padded) == 0:
        return padded

    # Find the first padding byte in the last 16 bytes
    end = padded.find(padded[-1], len(padded)-16, len(padded))

    # From the spec, the padding value should match the number of padded bytes.
    if len(padded) - end == padded[-1]:
        return padded[:end]

    return padded
```

Reply:
Not equivalent - the use of `find` here is error-prone, consider the case of:

b'foo\x01bar\x01bazquux\01'

Which should unpad to b'foo\x01bar\x01bazquux'

But your code will not unpad it, since it finds the first `\x01`.

Reply:
I'll see if I can make the code clearer though!